samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs

```diff
@@ -0,0 +1,254 @@
+    /// <param name="memoryStore">The memory store.</param>
+    /// <param name="chatRepository">The chat session repository.</param>
+    /// <param name="chatMessageRepository">The chat message repository.</param>
+    /// <param name="logger">The logger.</param>
+    public BotController(IConfiguration configuration, IMemoryStore memoryStore, ChatSessionRepository chatRepository, ChatMessageRepository chatMessageRepository, ILogger<SemanticKernelController> logger)
+    {
+        this._configuration = configuration;
+        this._logger = logger;
+        this._memoryStore = memoryStore;
+        this._chatRepository = chatRepository;
+        this._chatMessageRepository = chatMessageRepository;
+    }
+
+    /// <summary>
+    /// Upload a bot.
+    /// </summary>
+    /// <param name="kernel">The Semantic Kernel instance.</param>
+    /// <param name="userId">The user id.</param>
+    /// <param name="bot">The bot object from the message body</param>
+    /// <returns>The HTTP action result.</returns>
+    [Authorize]
+    [HttpPost]
+    [Route("bot/upload")]
+    [ProducesResponseType(StatusCodes.Status202Accepted)]
+    [ProducesResponseType(StatusCodes.Status415UnsupportedMediaType)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult> UploadAsync(
+        [FromServices] Kernel kernel,
+        [FromQuery] string userId,
+        [FromBody] Bot bot)
+    {
+        // TODO: We should get userId from server context instead of from request for privacy/security reasons when support multipe users.
+        this._logger.LogDebug("Received call to upload a bot");
+
+        if (!this.IsBotCompatible(bot.Schema, bot.EmbeddingConfigurations))
+        {
+            return new UnsupportedMediaTypeResult();
+        }
+
+        string chatTitle = $"{bot.ChatTitle} - Clone";
+        string chatId = string.Empty;
+
+        // Upload chat history into chat repository and embeddings into memory.
+        try
+        {
+            // 1. Create a new chat and get the chat id.
+            var newChat = new ChatSession(userId, chatTitle);
+            await this._chatRepository.CreateAsync(newChat);
+            chatId = newChat.Id;
+
+            string oldChatId = bot.ChatHistory.First().ChatId;
+
+            // 2. Update the app's chat storage.
+            foreach (var message in bot.ChatHistory)
+            {
+                var chatMessage = new ChatMessage(message.UserId, message.UserName, chatId, message.Content, ChatMessage.AuthorRoles.Participant);
+                chatMessage.Timestamp = message.Timestamp;
+                await this._chatMessageRepository.CreateAsync(chatMessage);
+            }
+
+            // 3. Update the memory.
+            await this.BulkUpsertMemoryRecordsAsync(oldChatId, chatId, bot.Embeddings);
+        }
+        catch
+        {
+            // TODO: Revert changes if any of the actions failed
+            throw;
```

Comment:
If we're not going to handle the errors, let's just have the TODO inline and remove the try/catch.

Reply:
I left the TODO there to handle the error in a separate PR.